From d4e16dab1e16f207a731eed1f79957a3eb9e8101 Mon Sep 17 00:00:00 2001
From: aken_liu <aken_liu@accton.com.tw>
Date: Fri, 12 Oct 2018 17:54:37 +0800
Subject: [PATCH] ACCTON-260: dco read value is strange Root casue: The access
 value '0' is a normal return data from DCO. We miss this case and do not
 update the DCO data in the kernel driver. Therefore, DHAL always gets a old
 value when the DCO data is 0.

Solution:
update the data when the value that gets from DCO is >= 0.
---
 drivers/misc/accton_t600_fj_mdec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/accton_t600_fj_mdec.c b/drivers/misc/accton_t600_fj_mdec.c
index 0391657..51a2b65 100644
--- a/drivers/misc/accton_t600_fj_mdec.c
+++ b/drivers/misc/accton_t600_fj_mdec.c
@@ -341,7 +341,7 @@ static ssize_t mdio_action_store(struct device* dev, struct device_attribute* at
     }
 
     status = mdio_access(dev, mode, page, address, data);
-    if((mode == MDIO_READ_MODE) && (status > 0))
+    if((mode == MDIO_READ_MODE) && (status >= 0))
     {
         mdio_read_result_data = status;
     }
-- 
1.9.1

